hive-llap/llap.sh

```diff
@@ -0,0 +1,497 @@
+    sed -i 's/long(max_direct_memory)/int(max_direct_memory)/g' /usr/lib/hive/scripts/llap/yarn/package.py
+}
+
+##if the metadata value exists, then we want to configure the local ssd as a caching location.
+function configure_SSD_caching_worker(){
+    if [[ -n "$HAS_SSD" ]]; then
+        echo "ssd"
+        mkdir /mnt/1/llap
+        chmod 777 /mnt/1/llap
+        chown hive:hive /mnt/1/llap
+        
+        bdconfig set_property \
+        --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+        --name 'hive.llap.io.allocator.mmap' --value 'true' \
+        --clobber
+        bdconfig set_property \
+        --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+        --name 'hive.llap.io.allocator.mmap.path' --value '/mnt/1/llap' \
+        --clobber
+        bdconfig set_property \
+        --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+        --name 'hive.llap.io.memory.mode' --value 'cache' \
+        --clobber
+        bdconfig set_property \
+        --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+        --name 'hive.llap.io.use.lrfu' --value 'true' \
+        --clobber
+    else
+        echo "NO SSD to configure..."
+    fi
+}
+
+##if the metadata value exists, then we want to configure the local ssd as a caching location. We don't need to make any directory changes since master nodes don't 
+##run YARN. We do need to ensure that the configuration files are equal across master and worker nodes
+function configure_SSD_caching_master(){
+    if [[ -n "$HAS_SSD" ]]; then
+       echo "ssd"
+
+        bdconfig set_property \
+        --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+        --name 'hive.llap.io.allocator.mmap' --value 'true' \
+        --clobber
+        bdconfig set_property \
+        --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+        --name 'hive.llap.io.allocator.mmap.path' --value '/mnt/1/llap' \
+        --clobber
+        bdconfig set_property \
+        --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+        --name 'hive.llap.io.memory.mode' --value 'cache' \
+        --clobber
+        bdconfig set_property \
+        --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+        --name 'hive.llap.io.use.lrfu' --value 'true' \
+        --clobber
+    else
+        echo "NO SSD to configure..."
+fi
+}
+
+##start LLAP - Master Node
+function start_llap(){
+
+    if [[ "${HOSTNAME}" == "${LLAP_MASTER_FQDN}" ]]; then
+
+    ##restart service after all configuration changes
+    echo "restart hive server prior..."
+    systemctl restart hive-server2.service 
+
+    echo "starting yarn app fastlaunch...."
+    yarn app -enableFastLaunch
+
+    echo "Setting Parameters for LLAP start"
+    
+    ###we want LLAP to have the entire YARN memory allocation for a node; easier to manage
+    LLAP_SIZE=$NODE_MANAGER_MEMORY
+    echo "LLAP daemon size: $LLAP_SIZE"
+
+    LLAP_CPU_ALLO=0
+    LLAP_MEMORY_ALLO=0
+    LLAP_XMX=0
+    LLAP_EXECUTORS=0
```

Comment:
if the variables are only used within the function, declare them as local variables. Like, `LLAP_EXECUTORS` -> `local llap_executors`

Reply:
done